src/main/java/net/sigmarik/abilitymod/mixin/PlayerEntityMixin.java

```diff
@@ -79,11 +85,48 @@ private void tickBoatMagnet() {
         }
     }
 
+    private void tickLightBurn() {
+        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_DAMAGED_BY_LIGHT)) {
+            World world = this.getWorld();
+            ItemStack helmet = this.getInventory().getArmorStack(3);
+            long time = world.getTimeOfDay() % 24000;
+            int lightLevel = world.getLightLevel(LightType.SKY, this.getBlockPos());
+            if (!(lightLevel <= 11 ||
+                    !world.isSkyVisible(this.getBlockPos()) || this.isTouchingWaterOrRain() || this.isInLava() ||
+                    this.hasStatusEffect(StatusEffect.byRawId(AbilityMod.FIRE_RESISTANCE_STATUS_EFFECT)) ||
+                    this.getBlockStateAtPos().getBlock().equals(Blocks.COBWEB) || (time >= 12542 && time < 23460) ||
+                    !world.getRegistryKey().getValue().equals(DimensionTypes.OVERWORLD_ID))) {
+                if (world.random.nextInt(AbilityMod.FIRING_CHANCE_PER_TICK) == 0) {
+                    if (helmet != null && helmet.isDamageable()) {
+                        this.getInventory().damageArmor((new DamageSources(world.getRegistryManager())).inFire(),
+                                1, PlayerInventory.HELMET_SLOTS);
+                    } else {
+                        this.setOnFireFromLava();
+                    }
+                }
+            }
+        }
+    }
+
+    private void tickIronBurn() {
+        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_HOT_IRON)) {
+            boolean doDamage = false;
+            for (int i = 0; i < 41; ++i) {
```

Comment:
It is time to start a witch hunt and burn every magical number.

You can use `this.getInventory().size()` to get the size of the player's inventory.

And also, even though it is a little controversial, it would be cool if you renamed iterator to something like `stackId` instead of single-character `i`.